sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/KeyVaultAccessControlAsyncClient.java

```diff
@@ -1044,4 +1045,26 @@ public T getValue() {
             return output;
         }
     }
+
+    /**
+     * Deserializes a given {@link Response HTTP response} including headers to a given class.
+     *
+     * @param statusCode The status code which will trigger exception swallowing.
+     * @param httpResponseException The {@link HttpResponseException} to be swallowed.
+     * @param logger {@link ClientLogger} that will be used to record the exception.
+     * @param <E> The class of the exception to swallow.
+     *
+     * @return A {@link Mono} that contains the deserialized response.
+     */
+    static <E extends HttpResponseException> Mono<Response<Void>> swallowExceptionForStatusCode(int statusCode,
+                                                                                                E httpResponseException, ClientLogger logger) {
+        HttpResponse httpResponse = httpResponseException.getResponse();
+
+        if (httpResponse.getStatusCode() == statusCode) {
+            return Mono.just(new SimpleResponse<>(httpResponse.getRequest(), httpResponse.getStatusCode(),
```

Comment:
Should the status code be `204` if we are not throwing an exception when we get a 404? The user will look at the status code and see 404 but the API returns successfully without throwing an exception. It would confuse the user.

Reply:
I think we are still leaving the status code as `404`. Is this correct @heaths?

Reply:
I would leave the 404. Not throwing for a 404 is one thing, but altering the actual response feels very wrong. The service returns a 404, even though it should've been returning a 204 like @johanste is pushing for newer REST APIs; however, keep in mind that Managed HSM intentionally copied the ARM RBAC design and behaviors here for the express purpose of making them consistent, as well as any client code expecting consistency.

Reply:
Talked to Heath directly and turns out that .NET is keeping the `404`. @annatisch also thinks we should not modify the status code returned by the service.

Reply:
We will not modify status codes.